modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectEntryServiceTest.java

```diff
@@ -718,29 +733,106 @@ public void testValidateMaximumNumberOfGuestUserObjectEntriesPerObjectDefinition
 			Date.from(
 				LocalDate.now(
 				).minusDays(
-					7
+					1
 				).atStartOfDay(
 					ZoneId.systemDefault()
 				).toInstant()));
 
-		_objectEntryLocalService.addOrUpdateObjectEntry(
-			objectEntry.getExternalReferenceCode(), _user.getUserId(),
-			TestPropsValues.getGroupId(),
-			_objectDefinition.getObjectDefinitionId(), objectEntry.getValues(),
+		objectEntry = _objectEntryLocalService.updateObjectEntry(objectEntry);
+
+		Assert.assertNotNull(
+			_objectEntryService.addObjectEntry(
+				0, _objectDefinition.getObjectDefinitionId(),
+				Collections.emptyMap(),
+				ServiceContextTestUtil.getServiceContext(
+					TestPropsValues.getGroupId(), _guestUser.getUserId())));
+
+		_objectEntryLocalService.deleteObjectEntry(
+			objectEntry.getObjectEntryId());
+
+		ConfigurationTestUtil.deleteConfiguration(
+			ObjectConfiguration.class.getName());
+
+		objectConfigurationDictionary.put(
+			"maximumNumberOfGuestUserObjectEntriesPerObjectDefinition", 3);
+		objectConfigurationDictionary.put("timeScale", "weeks");
+
+		ConfigurationTestUtil.saveConfiguration(
+			ObjectConfiguration.class.getName(), objectConfigurationDictionary);
+
+		objectEntry = _objectEntryService.addObjectEntry(
+			0, _objectDefinition.getObjectDefinitionId(),
+			Collections.emptyMap(),
 			ServiceContextTestUtil.getServiceContext(
 				TestPropsValues.getGroupId(), _guestUser.getUserId()));
 
+		objectEntry.setCreateDate(
+			Date.from(
+				LocalDate.now(
+				).minusDays(
+					7
+				).atStartOfDay(
+					ZoneId.systemDefault()
+				).toInstant()));
+
+		objectEntry = _objectEntryLocalService.updateObjectEntry(objectEntry);
+
+		Assert.assertNotNull(
+			_objectEntryService.addObjectEntry(
+				0, _objectDefinition.getObjectDefinitionId(),
+				Collections.emptyMap(),
+				ServiceContextTestUtil.getServiceContext(
+					TestPropsValues.getGroupId(), _guestUser.getUserId())));
+
 		AssertUtils.assertFailure(
 			ObjectEntryCountException.class,
 			StringBundler.concat(
-				"Unable to exceed ", 2,
-				" guest object entries for object definition ",
-				_objectDefinition.getObjectDefinitionId()),
+				"The limit of guest entries for ", _objectDefinition.getLabel(),
+				" has been reached and will no longer be accepted. Please ",
+				"contact the administrator for further assistance."),
 			() -> _objectEntryService.addObjectEntry(
 				0, _objectDefinition.getObjectDefinitionId(),
 				Collections.emptyMap(),
 				ServiceContextTestUtil.getServiceContext(
 					TestPropsValues.getGroupId(), _guestUser.getUserId())));
+
+		for (long adminUserId : adminUserIds) {
+			Assert.assertEquals(
+				1,
+				_userNotificationLocalService.getUserNotificationEventsCount(
```

Comment:
In the scenario above you used the _userNotificationLocalService.getDeliveredUserNotificationEvents method to check for notifications, so it seems to me that to maintain consistency you should also use the getDeliveredUserNotificationEventsCount method here, am I right?